AssertionError: Weightings not defined for all snapshots

## Checklist

- [x] I am using the current [`master`](https://github.com/pypsa/pypsa/tree/master) branch.

## Describe the Bug

To reproduce: clone the `pypsa-eur` repository (master) and try to run the tutorial. Make sure that the master version of PyPSA is installed in the `pypsa-eur` conda environment.

Backtrace:
```
rule prepare_network:
    input: networks/elec_s_5_ec.nc, data/costs.csv
    output: networks/elec_s_5_ec_lcopt_Co2L-24H.nc
    log: logs/prepare_network/elec_s_5_ec_lcopt_Co2L-24H.log
    jobid: 2
    benchmark: benchmarks/prepare_network/elec_s_5_ec_lcopt_Co2L-24H
    wildcards: simpl=, clusters=5, ll=copt, opts=Co2L-24H
    resources: mem=4000

INFO:pypsa.io:Imported network elec_s_5_ec.nc has buses, carriers, generators, lines, links, loads, stores
INFO:__main__:N-1 security margin of lines set to 0.7
INFO:__main__:Resampling the network to 24H
Traceback (most recent call last):
  File "/home/<user>/tmp/pypsa-eur/.snakemake/scripts/tmpuo3_haro.prepare_network.py", line 238, in <module>
    n = average_every_nhours(n, m.group(0))
  File "/home/<user>/tmp/pypsa-eur/.snakemake/scripts/tmpuo3_haro.prepare_network.py", line 144, in average_every_nhours
    m = n.copy(with_time=False)
  File "[...]/PyPSA/pypsa/components.py", line 872, in copy
    network.snapshot_weightings = self.snapshot_weightings.copy()
  File "[...]/PyPSA/pypsa/components.py", line 434, in snapshot_weightings
    assert df.index.equals(self.snapshots), "Weightings not defined for all snapshots."
AssertionError: Weightings not defined for all snapshots.
```
So something is going wrong in copying a network without time information (`with_time=False`). The relevant part in `pypsa/components.py` is: (line 862)
```
        if with_time:
            [...]
        else:
            network.snapshot_weightings = self.snapshot_weightings.copy()
```
This goes wrong because the setter for `snapshot_weightings` checks that the index of the weightings is the same as the index of the snapshots: (line 434)
```
assert df.index.equals(self.snapshots), "Weightings not defined for all snapshots."
```

I'm not quite sure what the best solution is. Is it the intention that the `snapshot_weightings` are copied when `with_time == False`?

--- a/doc/release_notes.rst
+++ b/doc/release_notes.rst
@@ -35,6 +35,8 @@ Upcoming Release
 
 * Bugfix in ``network.ilopf()`` where previously all links were fixed in the final iteration when it should only be the DC links.
 
+* ``n.snapshot_weightings`` is no longer copied for ``n.copy(with_time=False)``.
+
 
 PyPSA 0.17.1 (15th July 2020)
 =============================

--- a/pypsa/components.py
+++ b/pypsa/components.py
@@ -868,8 +868,6 @@ class Network(Basic):
                 for k in component.pnl.keys():
                     pnl[k] = component.pnl[k].loc[snapshots].copy()
             network.snapshot_weightings = self.snapshot_weightings.loc[snapshots].copy()
-        else:
-            network.snapshot_weightings = self.snapshot_weightings.copy()
 
         #catch all remaining attributes of network
         for attr in ["name", "srid"]:

